src/framework/preferences/js/ArrowScrolling.js

```diff
@@ -0,0 +1,119 @@
+            scrollContainer: ".flc-prefsEditor-scrollContainer"
+        },
+        model: {
+            // panelMaxIndex: null, // determined by the number of panels calculated after the onPrefsEditorMarkupReady event fired
+            // scrollToIndex: null, // the raw index set by translateToScroll, will be transformed to the panelIndex
+            panelIndex: 0
+        },
+        events: {
+            // beforeReset: null // should be provided by the fluid.prefs.prefsEditor grade
+        },
+        modelRelay: {
+            target: "panelIndex",
+            forward: {excludeSource: "init"},
+            singleTransform: {
+                type: "fluid.transforms.limitRange",
+                input: "{that}.model.scrollToIndex",
+                min: 0,
+                max: "{that}.model.panelMaxIndex"
+            }
+        },
+        modelListeners: {
+            "panelIndex": {
+                listener: "fluid.prefs.arrowScrolling.scrollToPanel",
+                args: ["{that}", "{change}.value"]
+            }
+        },
+        listeners: {
+            "onReady.windowResize": {
+                "this": window,
+                method: "addEventListener",
+                args: ["resize", "{that}.events.onSignificantDOMChange.fire"]
+            },
+            // Need to set panelMaxIndex after onPrefsEditorMarkupReady to ensure that the template has been
+            // rendered before we try to get the number of panels.
+            "onPrefsEditorMarkupReady.setPanelMaxIndex": {
+                changePath: "panelMaxIndex",
+                value: {
+                    expander: {
+                        funcName: "fluid.prefs.arrowScrolling.calculatePanelMaxIndex",
+                        args: ["{that}.dom.panels"]
+                    }
+                }
+            },
+            "beforeReset.resetPanelIndex": {
+                listener: "{that}.applier.fireChangeRequest",
+                args: {path: "panelIndex", value: 0, type: "ADD", source: "reset"}
+            }
+        },
+        invokers: {
+            scrollToPanel: {
+                changePath: "scrollToIndex",
+                value: "{arguments}.0"
+            },
+            translateToScroll: {
+                funcName: "fluid.prefs.arrowScrolling.translateToScroll",
+                args: ["{that}", "{arguments}.0"]
+            }
+        },
+        distributeOptions: [{
+            record: {
+                "afterRender.bindScrollArrows": {
+                    "this": "{that}.dom.header",
+                    method: "click",
+                    args: ["{prefsEditor}.translateToScroll"]
+                }
+            },
+            target: "{that > fluid.prefs.panel}.options.listeners"
+        }]
+
+    });
+
+    fluid.prefs.arrowScrolling.calculatePanelMaxIndex = function (panels) {
+        return Math.max(0, panels.length - 1);
+    };
+
+    fluid.prefs.arrowScrolling.translateToScroll = function (that, event) {
```

Comment:
Slightly better naming for this function/invoker - something like "eventToScrollIndex"?